### PR TITLE
2.0.2 Release - Add 'Recommended Version' API & User Public Command

### PR DESCRIPTION
**Pull request by nagilson**
- Bumps version numbers for 2.0.2
- Adds the ability for users to call the global SDK command themselves. Adds a gif below to show how to use it to the main page.
- Adds some documentation about our distro support policy and how to contribute.
- Moves the recommended version and list versions API out of the SDK and to the Runtime extension since we changed our plans with DevKit. Also I never fully implemented the piece which determines which linux version is in support to suggest, so I added it here.

Video demonstrating use of the command pallet to install .NET.

### CODE REVIEW

**Review by sa50511 — APPROVED** [review]

File: vscode-dotnet-runtime-extension/README.md [hunk 1]
```diff
@@ -17,6 +17,20 @@ This extension was probably included as a dependency of one of the following ext
 
 The above extensions call into this extension to provide a unified way of downloading shared .NET Runtimes or .NET SDKs. If you already have an installation of .NET that you'd like to use, see [the troubleshooting section below](#i-already-have-a-net-runtime-or-sdk-installed-and-i-want-to-use-it). If you want to remove this extension completely, you will need to uninstall any extensions that depend on it first. If this extension is uninstalled, any .NET Runtimes installed by it will also be removed.
 
+## Using the extension yourself
+
+As of version 2.0.2, you can install the .NET SDK using part of our private API via the VS Code Command Pallete!
```
Comment (nagilson) [inline vscode-dotnet-runtime-extension/README.md:22]:
```suggestion
As of version 2.0.2, you can install the .NET SDK using part of our private API via the VS Code Command Pallet!
```

File: vscode-dotnet-runtime-extension/README.md [hunk 2]
```diff
@@ -17,6 +17,20 @@ This extension was probably included as a dependency of one of the following ext
 
 The above extensions call into this extension to provide a unified way of downloading shared .NET Runtimes or .NET SDKs. If you already have an installation of .NET that you'd like to use, see [the troubleshooting section below](#i-already-have-a-net-runtime-or-sdk-installed-and-i-want-to-use-it). If you want to remove this extension completely, you will need to uninstall any extensions that depend on it first. If this extension is uninstalled, any .NET Runtimes installed by it will also be removed.
 
+## Using the extension yourself
+
+As of version 2.0.2, you can install the .NET SDK using part of our private API via the VS Code Command Palette!
+
+Bring up the command pallete (ctrl + shift + p) and run the command:
```
Comment (nagilson) [inline vscode-dotnet-runtime-extension/README.md:24]:
```suggestion
Bring up the command palette (ctrl + shift + p) and run the command:
```

File: Documentation/adding-distros.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# Adding Custom Distros
+
+Our initial support status for automated .NET installs on Linux includes Ubuntu and Red Hat Enterprise Linux. We realize this is a limited number of distros and made this chose based on the percentage of our users who would be impacted by this.
```
Comment (nagilson) [inline Documentation/adding-distros.md:3]:
```suggestion
Our initial support for automated .NET installs on Linux includes Ubuntu and Red Hat Enterprise Linux distros. We realize this is a limited number of distros.

```

File: vscode-dotnet-runtime-extension/README.md [hunk 4]
```diff
@@ -17,6 +17,20 @@ This extension was probably included as a dependency of one of the following ext
...
+
+As of version 2.0.2, you can install the .NET SDK using part of our private API via the VS Code Command Palette!
+
+Bring up the command palette (ctrl + shift + p) and run the command:
+.NET Install Tool - Install the .NET SDK System-Wide.
+
+![Video demonstrating use of the command pallet to install .NET.](https://raw.githubusercontent.com/dotnet/vscode-dotnet-runtime/63b7fca6c714781dc4cb1cdbcb786013f2115098/Documentation/example.gif)
+
+The command will try to find the best version of .NET for you to install, but you can tell it to install other versions as well based on its prompt.
+Note this feature is in-preview, and does not support all distros, WSL, nor preview versions of .NET.
```
Comment (nagilson) [inline vscode-dotnet-runtime-extension/README.md:30]:
```suggestion
Note this feature is in preview, and does not support all distros, WSL, nor preview or RC versions of .NET.
```

**Comment by nagilson** [comment]
@baronfel May you look at this please, especially from a 'PM' side? It's important to get the distro contribution document and the change to the main extension page worded right.

File: Documentation/adding-distros.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# Adding Custom Distros
+
+Our initial support for automated .NET installs on Linux includes Ubuntu and Red Hat Enterprise Linux distros. We realize this is a limited number of distros.
```
Comment (baronfel) [inline Documentation/adding-distros.md:3]:
```suggestion
Our initial support for automated .NET installs on Linux includes Ubuntu and Red Hat Enterprise Linux distros. Microsoft has first-party support for these distros as well as a few others, which you can learn more about at our [compatibility documentation](https://github.com/dotnet/core/blob/main/release-notes/8.0/supported-os.md#linux).

```

File: Documentation/adding-distros.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
+# Adding Custom Distros
+
+Our initial support for automated .NET installs on Linux includes Ubuntu and Red Hat Enterprise Linux distros. We realize this is a limited number of distros.
+
+
+While we plan on adding more distros in the future to meet the same support policy as VS Code, (which is also rather limited), we are happy to review PRs that add support for more distros. Only a few changes are required, which we will describe here. We are open to hearing your feedback on https://github.com/dotnet/vscode-dotnet-runtime/issues.
```
Comment (baronfel) [inline Documentation/adding-distros.md:6]:
```suggestion
While we plan on adding more distros in the future to meet the same support policy as [VS Code](https://code.visualstudio.com/docs/setup/linux), we are happy to review PRs that add support for more distros. Only a few changes are required, which we will describe here. We are open to hearing your feedback on https://github.com/dotnet/vscode-dotnet-runtime/issues.

```

**Comment by nagilson** [comment]
Added a preview feature flag to gate the feature and confirmed the command does not show up on WSL even if it is enabled.

File: vscode-dotnet-runtime-extension/README.md [hunk 7]
```diff
@@ -17,6 +17,23 @@ This extension was probably included as a dependency of one of the following ext
 
 The above extensions call into this extension to provide a unified way of downloading shared .NET Runtimes or .NET SDKs. If you already have an installation of .NET that you'd like to use, see [the troubleshooting section below](#i-already-have-a-net-runtime-or-sdk-installed-and-i-want-to-use-it). If you want to remove this extension completely, you will need to uninstall any extensions that depend on it first. If this extension is uninstalled, any .NET Runtimes installed by it will also be removed.
 
+## [Preview ]Using the extension yourself
```
Comment (nagilson) [inline vscode-dotnet-runtime-extension/README.md:20]:
```suggestion
## [Preview] Using the extension yourself
```

**Review by baronfel — APPROVED** [review]
Nice, that context hook-up looks almost exactly like I hoped it would. Glad that worked out!

File: vscode-dotnet-runtime-extension/src/test/functional/DotnetCoreAcquisitionExtension.test.ts [hunk 8]
```diff
@@ -235,6 +263,46 @@ suite('DotnetCoreAcquisitionExtension End to End', function() {
...
+
+    const result = await vscode.commands.executeCommand<string>('dotnet.recommendedVersion', null, webWorker);
+    assert.exists(result);
+    if(os.platform() !== 'linux')
+    {
+      assert.equal(result, '7.0.202', 'The SDK did not recommend the version it was supposed to, which should be {7.0.200} from the mock data.');
+    }
+    else
+    {
+      assert.equal(result, '7.0.1xx', 'The SDK did not recommend the version it was supposed to, which should be N.0.1xx based on surface level distro knowledge.');
```
Comment (nagilson) [inline vscode-dotnet-runtime-extension/src/test/functional/DotnetCoreAcquisitionExtension.test.ts:301]:
```suggestion
      assert.equal(result, '8.0.1xx', 'The SDK did not recommend the version it was supposed to, which should be N.0.1xx based on surface level distro knowledge. If a new version is available, this test may need to be updated to the newest version.');

```

**Pull request merged**